### PR TITLE
remarkable2-toolchain: init at 2.5.2

### PR DESCRIPTION
**Pull request by tadfisher**
###### Motivation for this change

Add the SDK for the reMarkable 2 tablet.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, libarchive, python3, file }:
+
+stdenv.mkDerivation rec {
+  pname = "remarkable-toolchain";
+  version = "2.5.2";
+
+  src = fetchurl {
+    url = "https://storage.googleapis.com/codex-public-bucket/codex-x86_64-cortexa7hf-neon-rm11x-toolchain-2.5.2.sh";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix:8]:
```suggestion
    url = "https://storage.googleapis.com/codex-public-bucket/codex-x86_64-cortexa7hf-neon-rm11x-toolchain-${version}.sh";
```

File: pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "1v410q1jn8flisdpkrymxd4pa1ylawd0rh3rljjpkqw1bp8a5vw1";
+  };
+
+  nativeBuildInputs = [
+    libarchive
+    python3
+    file
+  ];
+
+  unpackCmd = "mkdir src; install $curSrc src/install-toolchain.sh";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix:18]:
```suggestion
  unpackCmd = ''
    mkdir src
    install $curSrc src/install-toolchain.sh
  '';
```

File: pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    sed -i -e '3,9d' install-toolchain.sh # breaks PATH
+    sed -i 's|PYTHON=.*$|PYTHON=${python3}/bin/python|' install-toolchain.sh
+    ./install-toolchain.sh -D -y -d $out
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A toolchain for cross-compiling to reMarkable 2 tablets";
+    homepage = "https://remarkable.engineering/";
+    license = licenses.gpl2;
+    maintainers = [ maintainers.tadfisher ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix:33]:
```suggestion
    maintainers = with maintainers; [ tadfisher ];
```

File: pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchurl, libarchive, python3, file }:
+
+stdenv.mkDerivation rec {
+  pname = "remarkable-toolchain";
```
Comment (lukegb) [inline pkgs/development/tools/misc/remarkable/remarkable2-toolchain/default.nix:4]:
```suggestion
  pname = "remarkable2-toolchain";
```

**Review by lukegb — CHANGES_REQUESTED** [review]

**Review by lukegb — APPROVED** [review]
LGTM: This... installed *something*.

I suspect, however, you probably want to fix the fixup phase, because I'm concerned it's doing weird things to the target sysroot.

But... this is the same as what the remarkable 1 toolchain does, so LGTM.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103030` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>remarkable2-toolchain</li>
  </ul>
</details>
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

**Pull request merged**